crates/biome_js_analyze/src/lint/nursery/no_floating_promises.rs

```diff
@@ -598,3 +633,185 @@ fn is_expression_a_promise(expr: &AnyJsExpression, model: &SemanticModel) -> boo
 
     true
 }
+
+/// Traverses up the syntax tree to find the class declaration and checks if a method is a promise.
+///
+/// This function traverses up the syntax tree from the given `JsThisExpression` to find the nearest
+/// class declaration. It then searches for a method or property in the class that matches the provided
+/// `target_name`. If a matching member is found, it checks if the member is a promise.
+///
+/// # Arguments
+///
+/// * `js_this_expression` - A `JsThisExpression` representing the `this` keyword in the syntax tree.
+/// * `target_name` - The name of the method or property to search for.
+/// * `model` - A reference to the `SemanticModel` used for resolving bindings.
+///
+/// # Returns
+///
+/// * `Some(true)` if the class member is a promise.
+/// * `Some(false)` if the class member is not a promise.
+/// * `None` if there is an error in the process or if the class member is not found.
+///
+/// # Examples
+///
+/// Example TypeScript code that would return `Some(true)`:
+/// ```typescript
+/// class Api {
+///   async returnsPromise(): Promise<string> {
+///     return 'value';
+///   }
+///   async someMethod() {
+///     this.returnsPromise();
+///   }
+/// }
+/// ```
+///
+/// Example TypeScript code that would return `Some(false)`:
+/// ```typescript
+/// class Api {
+///   returnsString(){
+///     return 'value';
+///   }
+///   async someMethod() {
+///     this.returnsString();
+///   }
+/// }
+/// ```
+fn check_this_expression(
+    js_this_expression: &JsThisExpression,
+    target_name: &str,
+    model: &SemanticModel,
+) -> Option<bool> {
+    js_this_expression
+        .syntax()
+        .ancestors()
+        .skip(1)
+        .find_map(|ancestor| {
+            if ancestor.kind() == JsSyntaxKind::JS_CLASS_DECLARATION {
```

Comment:
A class can be initialised as an a declaration or an expression. This means we have to cover expressions too. They have a different node. [Check the playground](https://biomejs.dev/playground/?lintRules=all&code=YwBsAGEAcwBzACAAQQAgAHsACgAgACAACgB9AAoACgBjAG8AbgBzAHQAIABiACAAPQAgAGMAbABhAHMAcwAgAHsACgAgACAACgB9AA%3D%3D):

```js
class A {}

const b = class {}
```

Reply:
I have added support to class initialized as an expression.

Also, I have added support for method calls from outside of the Class
```
class Test {
  async returnsPromise(): Promise<string> {
    return 'value';
  }
}
const test = new Test();
test.returnsPromise(); // diagnostic here
```

Reply:
Thank you. It seels that class expression aren't supported yet:

```js
const b = class {}
```

Do you plan to do that in this PR?

Reply:
Oh, I see! Now I understand what you mean.
I will do that tomorrow. If you want to merge this PR now, please go ahead, and I will create a separate PR for adding that support.

Reply:
@ematipico 

I created a PR! https://github.com/biomejs/biome/pull/5098
Support for unnamed class expressions in `noFloatingPromises` was already implemented when I added support for named class expressions, as shown below. However, the test cases were missing, so I have added test cases for unnamed class expressions.
```
const b = class B {}
```